src/components/autocomplete/js/autocompleteController.js

```diff
@@ -640,15 +640,11 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
    * Clears the searchText value and selected item.
    */
   function clearValue () {
-    // Set the loading to true so we don't see flashes of content.
-    // The flashing will only occour when an async request is running.
-    // So the loading process will stop when the results had been retrieved.
-    setLoading(true);
```

Comment:
@crisbeto Those flashes are appearing, when asynchronously querying. I actually wasn't the guy who introduced this fix, but it seems reasonable for me. 

We should keep that, and set the `loading` to false, once the results are fetched successfully. 
Everything works fine, when caching is disabled, so the issue is, that we're resetting the `loading`, when the results were already cached.

To fix the issue, we simply need to reset the loading [here](https://github.com/crisbeto/material/blob/fix/autocomplete-progress-again/src/components/autocomplete/js/autocompleteController.js#L772)